controller/login.go

```diff
@@ -157,11 +202,19 @@ func (c *LoginController) Refresh(ctx *app.RefreshLoginContext) error {
 	if err != nil {
 		log.Error(ctx, map[string]interface{}{
 			"err": err,
-		}, "Unable to get Keycloak token endpoint URL")
+		}, "unable to get Keycloak token endpoint URL")
 		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, errs.Wrap(err, "unable to get Keycloak token endpoint URL")))
 	}
 
-	rpt, err := auth.GetEntitlement(ctx, entitlementEndpoint, nil, *token.AccessToken)
+	resources, err := c.getEntitlementResourceRequestPayload(ctx, token.AccessToken)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "failed to obtain create entitlement resource request ")
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrInternal(err.Error()))
+	}
+
+	rpt, err := auth.GetEntitlement(ctx, entitlementEndpoint, resources, *token.AccessToken)
```

Comment:
What if "recentSpaces" contains some incorrect data? For example some deleted space.
Plus users can access spaces which where they are not collaborators. So there are many use cases when you won't get an RPT token here. Make sure this function will return nil, nil in all cases. If the RPT is nil then just refresh the access token and return it.
But if for example KC returns an error (other than forbidden which is handled properly in GetEntitlement()) when we are trying to obtain an RPT for unknown resource (deleted space) then we should handle it too.
We don't want to break token refresh in all these use cases.